src/export/bops/model.ts

```diff
@@ -0,0 +1,146 @@
+  };
+  [key: string]: LooseNode;
+};
+
+interface LooseCrumb extends Crumb {
+  data?: Record<string, any>;
+}
+
+export type LooseBreadcrumbs = {
+  [key: string]: LooseCrumb;
+}
+
+export const DEFAULT_APPLICATION_TYPE = "lawfulness_certificate";
+
+// See minimum POST schema for /api/v1/planning_applications
+// https://ripa.bops.services/api-docs/index.html
+interface BOPSMinimumPayload {
+  application_type: typeof DEFAULT_APPLICATION_TYPE | string;
+  site: {
+    uprn?: string;
+    address_1: string;
+    address_2?: string;
+    town?: string;
+    postcode?: string;
+    latitude: string;
+    longitude: string;
+    x: string;
+    y: string;
+    source: string;
+  };
+  applicant_email: string;
+}
+
+export interface BOPSFullPayload extends BOPSMinimumPayload {
+  description?: string;
+  payment_reference?: string;
+  payment_amount?: number;
+  ward?: string;
+  work_status?: "proposed" | "existing";
+  applicant_first_name?: string;
+  applicant_last_name?: string;
+  applicant_phone?: string;
+  agent_first_name?: string;
+  agent_last_name?: string;
+  agent_phone?: string;
+  agent_email?: string;
+  proposal_details?: Array<QuestionAndResponses>;
+  feedback?: {
+    result?: string;
+    find_property?: string;
+    planning_constraints?: string;
+  };
+  constraints?: Record<string, boolean>;
+  files?: Array<File>;
+  boundary_geojson?: unknown;
+  result?: {
+    flag?: string;
+    heading?: string;
+    description?: string;
+    override?: string;
+  };
+  planx_debug_data?: Record<string, unknown>;
+  // typeof arr[number] > https://steveholgado.com/typescript-types-from-arrays
+  user_role?: typeof USER_ROLES[number];
+  works?: {
+    start_date?: string;
+    finish_date?: string;
+  };
+}
+
+export const USER_ROLES = ["applicant", "agent", "proxy"] as const;
+
+export interface QuestionMetaData {
+  notes?: string;
+  auto_answered?: boolean;
+  policy_refs?: Array<{
+    url?: string;
+    text?: string;
+  }>;
+  portal_name?: string;
+  section_name?: string;
+  feedback?: string;
+}
+
+export interface ResponseMetaData {
+  flags?: Array<string>;
+}
+
+export interface Response {
+  value: string;
+  metadata?: ResponseMetaData;
+}
+
+export interface QuestionAndResponses {
+  question: string;
+  metadata?: QuestionMetaData;
+  responses: Array<Response>;
+}
+
+// Using PLAN_TAGS & EVIDENCE_TAGS provided by BOPS, from:
+// https://github.com/unboxed/bops/blob/master/app/models/document.rb
+// will also be in POST Schema https://ripa.bops.services/api-docs
+type PlanTag =
+  | "Front"
+  | "Rear"
+  | "Side"
+  | "Roof"
+  | "Floor"
+  | "Site"
+  | "Plan"
+  | "Elevation"
+  | "Section"
+  | "Proposed"
+  | "Existing";
+
+type EvidenceTag =
+  | "Photograph"
+  | "Utility Bill"
+  | "Building Control Certificate"
+  | "Construction Invoice"
+  | "Council Tax Document"
+  | "Tenancy Agreement"
+  | "Tenancy Invoice"
+  | "Bank Statement"
+  | "Statutory Declaration"
+  | "Other"
+  | "Sitemap";
```

Comment:
`Sitemap` doesn't appear in [BOP's tags](https://github.com/unboxed/bops/blob/master/app/models/document.rb) - is that new?

Reply:
Good catch, I think this is an outdated tag - now removed!